### PR TITLE
Fix/delivery comparison

### PR DESCRIPTION
**Pull request by kaisermann**
#### What is the purpose of this pull request?

Some packages have the same `shippingEstimate` (13 i.e) but with different `shippingEstimateDate`s. This causes the comparison in question to not always choose the worst case scenario.

#### What problem is this solving?

[See clubhouse story](https://app.clubhouse.io/vtex-dev/story/7842/oms-ui-mostra-a-pior-data-de-entrega-no-detalhe-do-oms)

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by kaisermann** [comment]
Currently, tests are not passing. This is a WIP to begin a discussion about the issue.

**Comment by kaisermann** [comment]
Tests passing now 👍

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@vtex/delivery-packages",
-  "version": "2.17.0",
+  "version": "2.17.1-beta",
```
Comment (guilhermebruzzi) [inline package.json]: Just generate a stable tag and erase the beta commit :D (We generally create a beta branch to deal with these tests)

File: tests/index.test.js [hunk 2]
```diff
@@ -180,22 +179,30 @@ describe('has one package and a delivery', () => {
...
-    const packageAttachment = {
-      packages: [
-        createPackage([
-          { itemIndex: 0, quantity: 1 },
-          { itemIndex: 1, quantity: 1 },
-        ]),
-      ],
-    }
-    const selectedAddresses = [residentialAddress]
+  const items = createItems(2)
```
Comment (guilhermebruzzi) [inline tests/index.test.js]: Is this inside an `it` statement? Im confused with git diff

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [2.17.1-beta] - 2019-07-04
```
Comment (salesfelipe) [inline CHANGELOG.md]: You will need to remove this before releasing it

**Review by guilhermebruzzi — CHANGES_REQUESTED** [review]

**Review by salesfelipe — APPROVED** [review]

File: tests/index.test.js (hunk 2 above)
Comment (kaisermann) [reply tests/index.test.js]: It is not. I've moved it to the `describe` scope to reuse the values on the other test. Do you prefer that I keep it inside the `it`scope?

File: package.json (hunk 1 above)
Comment (kaisermann) [reply package.json]: Will do!

File: CHANGELOG.md (hunk 3 above)
Comment (kaisermann) [reply CHANGELOG.md]: Yeeeees sir

File: tests/index.test.js (hunk 2 above)
Comment (guilhermebruzzi) [reply tests/index.test.js]: @kaisermann Yeah, I think is better to that all variables live on the `it` scope, so the tests can modify their dependencies easier if needed.

But it doesn't mean you cant create `const` values outside all `describe` context, like:
```js
const BASE_ITEMS = createItems(2)
const BASE_PACKAGE_ATTACHMENT = {
      packages: [
        createPackage([
          { itemIndex: 0, quantity: 1 },
          { itemIndex: 1, quantity: 1 },
        ]),
      ],
    }
    const SELECTED_ADDRESSES = [residentialAddress]
```

And on the `it` scope you do:
```js
const items = BASE_ITEMS
const packageAttachment = BASE_PACKAGE_ATTACHMENT
const selectedAddresses = SELECTED_ADDRESSES
```

What you think?

Comment (kaisermann) [reply tests/index.test.js]: The values vary a lot on other tests. I agree with you but I don't think it fits this scenario. I've just commited with the consts insided each `it`

**Review by guilhermebruzzi — APPROVED** [review]
Great, just make sure the README is updated

**Pull request merged**